spec/controllers/wishlist_controller_spec.rb

```diff
@@ -0,0 +1,158 @@
+require 'rails_helper'
+
+RSpec.describe WishlistController, type: :controller do
+  let(:cas_session) { {cas_user: 'bob'} }
+  let(:wishlist_item) { double('wishlist_item') }
+  let(:relation) { double(first_or_initialize: wishlist_item) }
+
+  describe 'POST #add_course' do
+    context 'when the item can be saved' do
+      before { allow(wishlist_item).to receive(:save).and_return true }
```

Comment:
oooooooooh rspec thing. Wow, you can tell I don't write tests in Ruby daily anymore. Sorry. Ignore these comments. For posterity's sake though: https://www.relishapp.com/rspec/rspec-mocks/v/2-99/docs/method-stubs/allow-with-a-simple-return-value